platform/core/src/services/MeasurementService/MeasurementService.js

```diff
@@ -0,0 +1,444 @@
+ * @property {string} name -
+ * @property {string} version -
+ */
+
+/**
+ * Measurement schema
+ *
+ * @typedef {Object} Measurement
+ * @property {number} id -
+ * @property {string} sopInstanceUID -
+ * @property {string} frameOfReferenceUID -
+ * @property {string} referenceSeriesUID -
+ * @property {string} label -
+ * @property {string} description -
+ * @property {string} type -
+ * @property {string} unit -
+ * @property {number} area -
+ * @property {Array} points -
+ * @property {MeasurementSource} source -
+ */
+
+/* Measurement schema keys for object validation. */
+const MEASUREMENT_SCHEMA_KEYS = [
+  'id',
+  'sopInstanceUID',
+  'frameOfReferenceUID',
+  'referenceSeriesUID',
+  'label',
+  'description',
+  'type',
+  'unit',
+  'area', // TODO: Add concept names instead (descriptor)
+  'points',
+  'source',
+];
+
+const EVENTS = {
+  MEASUREMENT_UPDATED: 'event::measurement_updated',
+  MEASUREMENT_ADDED: 'event::measurement_added',
+};
+
+const VALUE_TYPES = {
+  POLYLINE: 'value_type::polyline',
+  POINT: 'value_type::point',
+  ELLIPSE: 'value_type::ellipse',
+  MULTIPOINT: 'value_type::multipoint',
+  CIRCLE: 'value_type::circle',
+};
+
+class MeasurementService {
+  constructor() {
+    this.sources = {};
+    this.mappings = {};
+    this.measurements = {};
+    this.listeners = {};
+    Object.defineProperty(this, 'EVENTS', {
+      value: EVENTS,
+      writable: false,
+      enumerable: true,
+      configurable: false,
+    });
+    Object.defineProperty(this, 'VALUE_TYPES', {
+      value: VALUE_TYPES,
+      writable: false,
+      enumerable: true,
+      configurable: false,
+    });
+  }
+
+  /**
+   * Get all measurements.
+   *
+   * @return {Measurement[]} Array of measurements
+   */
+  getMeasurements() {
+    return this._arrayOfObjects(this.measurements);
+  }
+
+  /**
+   * Get specific measurement by its id.
+   *
+   * @param {string} id If of the measurement
+   * @return {Measurement} Measurement instance
+   */
+  getMeasurement(id) {
+    let measurement = null;
+    if (Object.keys(this.measurements[id]).length > 0) {
+      measurement = this.measurements[id];
+    }
+    return measurement;
+  }
+
+  /**
+   * Create a new source.
+   *
+   * @param {string} name Name of the source
+   * @param {string} version Source name
+   * @return {MeasurementSource} Measurement source instance
+   */
+  createSource(name, version) {
+    if (!name) {
+      log.warn('Source name not provided. Exiting early.');
+      return;
+    }
+
+    if (!version) {
+      log.warn('Source version not provided. Exiting early.');
+      return;
+    }
+
+    const id = guid();
+    const source = {
+      id,
+      name,
+      version,
+    };
+    source.addOrUpdate = (definition, measurement) => {
+      return this.addOrUpdate(source, definition, measurement);
+    };
+    source.getAnnotation = (definition, measurementId) => {
+      return this.getAnnotation(source, definition, measurementId);
+    };
```

Comment:
If we have these methods on source, do we want to expose `getAnnotation` and `addOrUpdate` where we take `source` has a param, or make them private?